optuna/study.py

```diff
@@ -454,6 +467,134 @@ def _log_completed_trial(self, trial_number, value):
 
 
+class RunningStudy(object):
+    """An object to access a running :class:`~optuna.study.Study`.
+
+    Note that this object is created within Optuna library, so
+    it is not intended that library users directly use this constructor.
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object.
+
+    """
+
+    def __init__(self, study):
+        # type: (Study) -> None
+
+        self.study_name = study.study_name
+        self.study_id = study.study_id
+        self.storage = study.storage
+
+    @property
+    def best_params(self):
+        # type: () -> Dict[str, Any]
+        """Return parameters of the best trial in the :class:`~optuna.study.RunningStudy`.
+
+        Returns:
+            A dictionary containing parameters of the best trial.
+        """
+
+        return self.best_trial.params
+
+    @property
+    def best_value(self):
+        # type: () -> float
+        """Return the best objective value in the :class:`~optuna.study.RunningStudy`.
+
+        Returns:
+            A float representing the best objective value.
+        """
+
+        best_value = self.best_trial.value
+        if best_value is None:
+            raise ValueError('No trials are completed yet.')
+
+        return best_value
+
+    @property
+    def best_trial(self):
+        # type: () -> structs.FrozenTrial
+        """Return the best trial in the :class:`~optuna.study.RunningStudy`.
+
+        Returns:
+            A :class:`~optuna.structs.FrozenTrial` object of the best trial.
+        """
+
+        return self.storage.get_best_trial(self.study_id)
+
+    @property
+    def direction(self):
+        # type: () -> structs.StudyDirection
+        """Return the direction of the :class:`~optuna.study.RunningStudy`.
+
+        Returns:
+            A :class:`~optuna.structs.StudyDirection` object.
+        """
+
+        return self.storage.get_study_direction(self.study_id)
+
+    @property
+    def trials(self):
+        # type: () -> List[structs.FrozenTrial]
+        """Return all trials in the :class:`~optuna.study.RunningStudy`.
+
+        Returns:
+            A list of :class:`~optuna.structs.FrozenTrial` objects.
+        """
+
+        return self.storage.get_all_trials(self.study_id)
+
+    @property
+    def system_attrs(self):
+        # type: () -> Dict[str, Any]
+        """Return system attributes.
+
+        Returns:
+            A dictionary containing all system attributes.
+        """
+
+        return self.storage.get_study_system_attrs(self.study_id)
+
+    def set_system_attr(self, key, value):
+        # type: (str, Any) -> None
+        """Set a system attribute to the :class:`~optuna.study.RunningStudy`.
+
+        Args:
+            key: A key string of the attribute.
+            value: A value of the attribute. The value should be JSON serializable.
+
+        """
+
+        self.storage.set_study_system_attr(self.study_id, key, value)
+
+    @property
+    def full_search_space(self):
```

Comment:
This method is not called anywhere in this PR (except for `test_full_search_space` unit test function).
I think we can remove this method from the PR (and will add it later when it is actually required).
What do you think? > @toshihikoyanase @g-votte

Reply:
I prefer removing `full_search_space` from this PR.

Reply:
Thank you. I will drop this method.

Reply:
Done: https://github.com/pfnet/optuna/pull/380/commits/4236a6b905edef5bb575a31bb662bb1387d80765